noty-android/app/simpleapp/src/main/java/dev/shreyaspatil/noty/simpleapp/view/base/BaseFragment.kt

```diff
@@ -47,12 +57,30 @@ abstract class BaseFragment<VB : ViewBinding, VM : ViewModel> : Fragment() {
         return binding.root
     }
 
+    abstract fun initView()
+    abstract fun render(state: STATE)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initView()
+        observeState()
+    }
+
+    private fun observeState() {
+        viewModel.state
+            .flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
```

Comment:
Since this is a Fragment, use Fragment View's lifecycle instead of Fragment's lifecycle
```suggestion
            .flowWithLifecycle(viewLifecycleOwner.lifecycle, Lifecycle.State.STARTED)
```